src/Aquifer.API/Endpoints/Resources/Content/SendForPublisherReview/Endpoint.cs

```diff
@@ -34,47 +70,85 @@ public override async Task HandleAsync(Request request, CancellationToken ct)
             ThrowError("One or more resources not found or not in correct status");
         }
 
-        var user = await userService.GetUserFromJwtAsync(ct);
-        var isPublisher = user.Role == UserRole.Publisher;
-        foreach (var draftVersion in draftVersions)
+        return draftVersions;
+    }
+
+    private async Task AssignUserToContent(ResourceContentStatus newStatus, int? assignedUserId, ResourceContentVersionEntity draftVersion,
+        UserEntity user, CancellationToken ct)
+    {
+        // there shouldn't be an assigned user if it's going into review pending
+        if (Constants.ReviewPendingStatuses.Contains(newStatus))
+        {
+            await historyService.AddAssignedUserHistoryAsync(draftVersion, null, user.Id, ct);
+            draftVersion.AssignedUserId = null;
+        }
+        else
         {
-            if (user.Id != draftVersion.AssignedUserId)
+            if (assignedUserId is not null && assignedUserId != draftVersion.AssignedUserId)
             {
-                ThrowError("User must be assigned to content to send for publisher review.");
+                await historyService.AddSnapshotHistoryAsync(draftVersion,
+                    draftVersion.AssignedUserId,
+                    draftVersion.ResourceContent.Status,
+                    ct);
+                await historyService.AddAssignedUserHistoryAsync(draftVersion, assignedUserId, user.Id, ct);
+                draftVersion.AssignedUserId = assignedUserId;
             }
+        }
+    }
 
-            var newStatus = GetNewStatus(isPublisher, draftVersion.ResourceContent.Status);
-
-            await historyService.AddSnapshotHistoryAsync(draftVersion,
-                draftVersion.AssignedUserId,
-                draftVersion.ResourceContent.Status,
-                ct);
+    private async Task ValidateAssignedUser(UserEntity user, int? assignedUserId, ResourceContentVersionEntity draftVersion,
+        CancellationToken ct)
+    {
+        if (user.Id != draftVersion.AssignedUserId && Constants.CompanyReviewStatuses.Contains(draftVersion.ResourceContent.Status))
+        {
+            ThrowError("User must be assigned to content to send for publisher review.");
+        }
 
-            Helpers.SanitizeTiptapContent(draftVersion);
-            draftVersion.ResourceContent.Status = newStatus;
-            if (!isPublisher)
+        if (assignedUserId is not null)
+        {
+            var assignedUser = await dbContext.Users
+                .AsTracking()
+                .SingleOrDefaultAsync(u => u.Id == assignedUserId && u.Enabled, ct);
```

Comment:
I'd just put the publisher role check here and call it good.

Reply:
I was meaning that you should check the Role here `&& u.Role == UserRole.Publisher` and then keep the null check on `assignedUser`.